File: frontend/src/types/property.ts
export type RiskLevel = "low" | "moderate" | "high" | "very_high" | "extreme";
export type Facade = "north" | "south" | "east" | "west";

export interface FacadeHeatScore {
  facade: Facade;
  heat_gain_score: number; // 0-100
  worst_month: number;
  best_month: number;
}

export interface CarHeatRisk {
  hour: number;
  outdoor_temp_c: number;
  estimated_interior_temp_c: number;
  risk_level: RiskLevel;
}

export interface PropertyAnalysis {
  location_id: string;
  facade_scores: FacadeHeatScore[];
  worst_car_heat_hour: CarHeatRisk;
  outdoor_comfort_score: number;
  solar_panel_score: number; // 0-100 potential
  annual_heat_risk: RiskLevel;
  monthly_comfort: { month: number; score: number }[];
}

File: backend/app/schemas/property.py
from pydantic import BaseModel


class FacadeHeatScoreOut(BaseModel):
    facade: str         # north / south / east / west
    heat_gain_score: float  # 0-100
    worst_month: int
    best_month: int


class CarHeatRiskOut(BaseModel):
    hour: int
    outdoor_temp_c: float
    estimated_interior_temp_c: float
    risk_level: str     # low / moderate / high / very_high / extreme


class PropertyAnalysisOut(BaseModel):
    location_id: int
    facade_scores: list[FacadeHeatScoreOut]
    worst_car_heat: CarHeatRiskOut
    outdoor_comfort_score: float
    solar_panel_score: float
    annual_heat_risk_level: str
    monthly_comfort: list[dict]     # [{ month: int, score: float }]

    model_config = {"from_attributes": True}
